src/pipeline.ml

```diff
@@ -110,6 +110,64 @@ module Packet_unikernel = struct
 end
 module Build_unikernel = Build.Make(Packet_unikernel)
 
+module Docker_context = struct
+    module Taridescom = Current_docker.Make(struct let docker_context = Some "staging.tarides.com" end)
+    
+
+    type service = [
+        | `Taridescom of string
+    ]
+
+    type build_info = {
+        service: service;
+        target: string;
+        dockerfile: string;
+        args: (string * string) list;
+    }
+
+    type deploy_info = {
+        service: service;
+    }
+
+    let name { service } = match service with
+    | `Taridescom service -> service
+
+
+    let generate (module D : Current_docker.S.DOCKER) ~dockerfile ~target ~args:_ src =
+        let+ image =
+            D.build ~dockerfile
+                ~label:target
+                ~pull:true
+                ~timeout
+                (`Git src)
+        in
+        D.Image.hash image
+
+    let build_image { service ; dockerfile ; target ; args } src =
+        let src = Current_git.fetch src in
+        let dockerfile = Current.return (`File (Fpath.v dockerfile)) in
+        match service with
+        | `Taridescom _ -> generate (module Taridescom) ~dockerfile ~target ~args src
+       
+
+    let build build_info ?additional_build_args:_ _repo src  = Current.ignore_value (build_image build_info src)
+
+    let tag_and_serve (module D : Current_docker.S.DOCKER) ~image name =
+        let tag = "deployer-" ^  name in
+        let image = Current.map D.Image.of_hash image in
+        Current.all [
+            D.tag ~tag image;
+            D.service ~name ~image ()
+        ]
+
+    let deploy build_info { service } ?additional_build_args:_ src =
+        let image = build_image build_info src in
+        match service with
+        | `Taridescom name -> tag_and_serve (module Taridescom) ~image name
+        
+end
+module Docker_context_build = Build.Make(Docker_context)
```

Comment:
The indentation on this new section different to the rest of the file. Please reformat it to be consistent.